File: algorithms/heapsort/lastHeapIndex.cpp
#include "iostream"
#include "vector"

size_t lastHeapIndex(std::vector<int> &vec) {
    if (vec.empty()) {
        return 0;
    }

    for (int i = 1; i < vec.size(); ++i) {
        size_t parent = (i - 1) / 2;
        if (vec[i] > vec[parent]) {
            return i - 1;
        }
    }
    
    return vec.size() - 1;
}

File: algorithms/heap/lastHeapIndex.py
def lastHeapIndex(arr: list) -> int:
    if not arr:
        return 0

    for i in range(1, len(arr)):
        parent = (i - 1) // 2
        if arr[i] > arr[parent]:
            return i - 1

    return len(arr) - 1
